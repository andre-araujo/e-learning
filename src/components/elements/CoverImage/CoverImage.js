import React from 'react';
import { string, node } from 'prop-types';

import {
  Container,
  Content,
} from './styles';

function CoverImage({
  imageUrl,
  children,
}) {
  return (
    <Container imageUrl={imageUrl}>
      <Content>
        {children}
      </Content>
    </Container>
  );
}

CoverImage.defaultProps = {
  imageUrl: '#',
  children: null,
};

CoverImage.propTypes = {
  imageUrl: string,
  children: node,
};

export default CoverImage;
