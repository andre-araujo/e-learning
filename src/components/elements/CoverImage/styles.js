import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ imageUrl }) => imageUrl && css`
    background-image: url(${imageUrl});
    background-size: cover;
    background-position: center;
  `}

  position: relative;

  :before {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: rgba(0, 0, 0, .4)
  }
`;

export const Content = styled.div`
  position: relative;
`;
