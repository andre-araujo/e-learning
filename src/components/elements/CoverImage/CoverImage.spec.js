import React from 'react';
import { shallow } from 'enzyme';

import CoverImage from './CoverImage';

describe('CoverImage component', () => {
  it('should match snapshot', () => {
    expect(shallow(<CoverImage />)).toMatchSnapshot();
  });
});
