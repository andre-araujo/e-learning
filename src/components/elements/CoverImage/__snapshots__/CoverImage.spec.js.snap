// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImage component should match snapshot 1`] = `
<styled.div
  imageUrl="#"
>
  <styled.div />
</styled.div>
`;
